point_viewer_grpc/src/bin/octree_server.rs

```diff
@@ -64,6 +64,61 @@ impl proto_grpc::Octree for OctreeService {
         ctx.spawn(f)
     }
 
+    fn get_points_in_frustum(
+        &self,
+        ctx: RpcContext,
+        req: proto::GetPointsInFrustumRequest,
+        resp: ServerStreamingSink<proto::GetPointsInFrustumReply>,
+    ) {
+        let projection_matrix = Matrix4::from(PerspectiveFov {
+            fov_y: req.fov_y,
+            aspect: req.aspect,
+            z_near:req.z_near,
+            z_far: req.z_far,
+        });
+        let view_position = Point3::new(req.view_position.x, req.view_position.y, req.view_position.z);
+        let view_direction = Vector3::new(req.view_direction.x, req.view_direction.y, req.view_direction.z);
+        let view_up = Vector3::new(req.view_up.x, req.view_up.y, req.view_up.z);
+        let view_center = view_position + view_up;
+
+        let view_transform = Transform.look_at(view_position, view_center, view_up);
+        let view_matrix: Matrix4<f32> = view_transform.inverse_transform().unwrap().into();
+        let frustum_matrix = projection_matrix * view_matrix;
+
+        let octree = self.octree.clone();
```

Comment:
I think this is fine for the first PR. Later, we should extract everything from here into a function that accepts the matrix and returns the points. This way we can unify the code path for Frustum and AABB.

Reply:
Thanks for the help and review @thomasschiwietz

Reply:
Please add a TODO() at this line: Extract get_points_from_transformtion() as back end for get_points_in_box() and get_points_in_frustum().